frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/utils.ts

```diff
@@ -55,3 +89,218 @@ export function checkIfCancelled(datasetStatus: DatasetUploadStatus): boolean {
     datasetStatus.upload_status === UPLOAD_STATUS.CANCELED
   );
 }
+
+export function useCheckCollectionPopulated({
+  invalidateCollectionQuery,
+  isNamePopulated,
+  upload_progress,
+}: {
+  invalidateCollectionQuery: () => void;
+  isNamePopulated: boolean;
+  upload_progress: number;
+}) {
+  useCheckCollection({
+    invalidateCollectionQuery,
+    shouldFetch: upload_progress === 1 && !isNamePopulated,
+  });
+}
+
+type FORMAT_KEYS =
+  | "conversion_loom_status"
+  | "conversion_anndata_status"
+  | "conversion_cxg_status"
+  | "conversion_rds_status";
+
+const CONVERSION_STATUS_FORMAT_KEYS = [
+  "conversion_loom_status",
+  "conversion_anndata_status",
+  "conversion_cxg_status",
+  "conversion_rds_status",
+] as FORMAT_KEYS[];
+
+export function useCheckCollectionFormatsPopulated({
+  invalidateCollectionQuery,
+  dataset,
+  datasetUploadStatus,
+}: {
+  invalidateCollectionQuery: () => void;
+  dataset: Dataset;
+  datasetUploadStatus: DatasetUploadStatus;
+}) {
+  let numOfConvertedFormats = 0;
+
+  for (const key of CONVERSION_STATUS_FORMAT_KEYS) {
+    if (datasetUploadStatus[key] !== CONVERSION_STATUS.CONVERTED) continue;
+
+    numOfConvertedFormats += 1;
+  }
+
+  let numOfAvailableFormats = 0;
+
+  if (dataset.dataset_deployments.length > 0) {
+    numOfAvailableFormats += 1;
+  }
+
+  numOfAvailableFormats = numOfAvailableFormats + dataset.dataset_assets.length;
+
+  useCheckCollection({
+    invalidateCollectionQuery,
+    shouldFetch: numOfConvertedFormats > numOfAvailableFormats,
+  });
+}
+
+const FETCH_COLLECTION_INTERVAL_MS = 5 * 1000;
+
+function useCheckCollection({
+  invalidateCollectionQuery,
+  shouldFetch,
+}: {
+  invalidateCollectionQuery: () => void;
+  shouldFetch: boolean;
+}) {
+  useEffect(() => {
+    let intervalId: number | undefined = undefined;
+
+    if (!intervalId && shouldFetch) {
+      intervalId = window?.setInterval(() => {
+        if (shouldFetch) {
+          invalidateCollectionQuery();
+        }
+      }, FETCH_COLLECTION_INTERVAL_MS);
+    }
+
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, [invalidateCollectionQuery, shouldFetch]);
+}
+
+export function useCancelDatasetStatusQuery({
+  datasetId,
+  isFailed,
+  isLoading,
+  queryCache,
+}: {
+  datasetId: string;
+  isFailed: boolean;
+  isLoading: boolean;
+  queryCache: QueryCache;
+}) {
+  useEffect(() => {
+    if (isFailed || !isLoading) {
+      queryCache.cancelQueries([USE_DATASET_STATUS, datasetId]);
+    }
+  }, [datasetId, isFailed, isLoading, queryCache]);
+}
+
+export function useUploadProgress({
```

Comment:
`useUploadProgress() ` and `useConversionProgress()` both use `useProgress()` under the hood!